src/nibpkg/compose.nim

```diff
@@ -6,15 +6,35 @@ type
     left, right: string
 
 proc retrieve_flanking_sequences_from_fai(fastaIdx: Fai, chrom: string,
-        start_pos : int, end_pos : int, flank: int): FlankSeq =
+        start_pos: int, end_pos: int, flank: int): FlankSeq =
   ## this function lacks a return
   result.left = fastaIdx.get(chrom, max(0, start_pos - flank), start_pos)
   result.right = fastaIdx.get(chrom, end_pos, end_pos + flank)
 
+proc compose(variant: Variant, right_flank: string,
+    left_flank: string): string =
 
+  var inner_seq: string
+  var variant_type: string
+  doAssert variant.info.get("SVTYPE", variant_type) == Status.OK
+  if variant_type == "DEL":
+    inner_seq = ""
+  elif variant_type == "INS":
+    inner_seq = variant.ALT[0]
+  elif variant_type == "INV":
+    raise newException(ValueError,
+    "Error: Inversion processing not implemented.")
```

Comment:
nice.

Reply:
Every dang time - we'll implement those later!